### ISSUE TITLE
Circle CI broken due to flow errors

### ISSUE THREAD

**Issue by dlowder-salesforce** [facebook/react-native#15683]
### Is this a bug report?

Yes

### Have you read the [Contributing Guidelines](https://facebook.github.io/react-native/docs/contributing.html)?

Yes

### Environment

1. `react-native -v`: Current facebook/react-native#master
2. `node -v`: 7.5.0
3. `npm -v`: 4.1.2
4. `yarn --version`: 0.27.5

Then, specify:

- Target Platform: iOS, Apple TV

- Development Operating System: macOS

- Build tools: Xcode

### Steps to Reproduce

1. `git clone https://github.com/facebook/react-native`
2. `cd react-native`
3. `npm install` or `yarn`
4. `npm run flow check`

### Expected Behavior

The flow check should succeed.

### Actual Behavior

```
Error: Libraries/Experimental/WindowedListView.js:704
704:     this._containerRef = ref;
         ^^^^^^^^^^^^^^^^^^ property `setNativeProps`. Property not found in
704:     this._containerRef = ref;
                              ^^^ React$Component


... 87 more errors (only 50 out of 137 errors displayed)
To see all errors, re-run Flow with --show-all-errors

npm ERR! Darwin 16.7.0
npm ERR! argv "/usr/local/Cellar/node/7.5.0/bin/node" "/usr/local/bin/npm" "run" "flow" "check"
npm ERR! node v7.5.0
npm ERR! npm  v4.1.2
npm ERR! code ELIFECYCLE
npm ERR! react-native@1000.0.0 flow: `flow "check"`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the react-native@1000.0.0 flow script 'flow "check"'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the react-native package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     flow "check"
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs react-native
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls react-native
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/dlowder/Documents/iosProjects/react-native-commit/npm-debug.log
```

### Reproducible Demo

See above

**Comment by hramos** [comment]
Looking into it right now, thanks for opening an issue.

**Comment by hramos** [comment]
Recent changes that may have caused this: we upgraded to Flow v0.53 along with a big codemod recently.

**Comment by hramos** [comment]
The failure was introduced in https://github.com/facebook/react-native/commit/90eaeb019b2056150cae541a59720daad21466ef.

**Comment by hramos** [comment]
A fix has been proposed and if all goes well, it could land later today.

**Comment by hramos** [comment]
Fix landed in 63f990121ad5f58dfd07ccfed2bb640b08164f73

Circle hasn't been triggered by recent commits for some reason, but I was able to trigger a manual run at https://circleci.com/gh/facebook/react-native/20572. It's currently running, I'll check back on it later and close this issue if the Flow step finishes successfully

**Comment by hramos** [comment]
Only a couple errors left to fix now:

```
npm run flow -- check

> react-native@1000.0.0 flow /home/ubuntu/react-native
> flow "check"

Error: local-cli/server/util/attachHMRServer.js:56
 56:   getDependencies(options: DependencyOptions): Promise<ResolutionResponse<TModule>>,
                                ^^^^^^^^^^^^^^^^^ property `rootEntryFile`. Property not found in
351:     options: DependencyOptions,
                  ^^^^^^^^^^^^^^^^^ object type. See: node_modules/metro-bundler/src/Server/index.js.flow:351

Error: local-cli/server/util/attachHMRServer.js:58
 58:   getShallowDependencies(options: DependencyOptions): Promise<Array<TModule>>,
                                       ^^^^^^^^^^^^^^^^^ property `rootEntryFile`. Property not found in
328:   getShallowDependencies(options: DependencyOptions): Promise<Array<Module>> {
                                       ^^^^^^^^^^^^^^^^^ object type. See: node_modules/metro-bundler/src/Server/index.js.flow:328
```

**Comment by hramos** [comment]
Waiting on a new version of `metro-bundler` to be published.

**Comment by hramos** [comment]
This was fixed earlier today when `metro-bundler` 0.13.0 shipped. Unfortunately, another update to `attachHMRServer.js` landed an hour later, which will require another `metro-bundler` release to fix the new inconsistencies.

**Comment by hramos** [comment]
Solved. Remaining error is unrelated to Flow, but someone is already assigned to work on a fix.
